src/Systems.jl

```diff
@@ -4,34 +4,44 @@ import Random
 Defines a collection of spins, as well as the Hamiltonian they interact under.
  This is the main type to interface with most of the package.
 """
-struct SpinSystem{N}
+mutable struct SpinSystem{N}   ## Note: Changed to mutable so old integrator would work
     lattice     :: Lattice                          # Definition of underlying lattice
     hamiltonian :: HamiltonianCPU                   # Contains all interactions present
     _dipoles    :: Array{Vec3, 4}                   # Holds dipole moments: Axes are [Basis, CellA, CellB, CellC]
     _coherents  :: Array{SVector{N, ComplexF64}, 4} # Coherent states
     site_infos  :: Vector{SiteInfo}                 # Characterization of each basis site
+    S           :: NTuple{3, Matrix{ComplexF64}}
 end
 
 @inline Base.size(sys::SpinSystem) = size(sys._coherents)
 @inline nbasis(sys::SpinSystem) = nbasis(sys.lattice)
 @inline eachcellindex(sys::SpinSystem) = eachcellindex(sys.lattice)
 
+@inline _get_coherent_from_dipole(dip::Vec3, ::Val{0}) :: CVec{0} = CVec{0}(zeros(0))
 @inline function _get_coherent_from_dipole(dip::Vec3, ::Val{N}) :: CVec{N} where {N} 
-    # TODO: Produce a coherent state of SU(N) agreeing with the given dipole moment.
-    return CVec{N}(zeros(N))
+    S = gen_spin_ops(N) 
+    λs, vs = eigen(dip⋅S)
+    return CVec{N}(vs[:, argmax(real.(λs))])
 end
 
+# Construct an analogous CoherentView?
```

Comment:
Add a KetView?

Reply:
Added a KetView. Also realized there was a real issue with the indexing (which I dealt with before by writing `getindex` for both Int and VarArg arguments). This can be overcome by splatting the `i` argument on the generic `getindex` function.